src/components/Scorecards/ScorecardDetailsPage/ScorecardDetailsPage.test.tsx

```diff
@@ -282,7 +349,23 @@ describe('ScorecardDetailsPage', () => {
     );
     expect(await screen.findByText(/Failing Rule/)).toBeVisible();
     expect(await screen.queryByText(/63%/)).not.toBeInTheDocument();
-    expect((await screen.findAllByText(/50%/))[0]).toBeVisible();
+    expect((await screen.findAllByText(/38%/))[0]).toBeVisible(); // average score of 2 services fulfilling the filter
+    expect(await screen.findByText(/lorem/)).toBeVisible();
+    expect(await screen.findByText(/ipsum/)).toBeVisible();
+    expect(await screen.queryByText(/foo/)).not.toBeInTheDocument();
+
+    // Clear the filters for failed rule and check exempt rules
+    fireEvent.click(await screen.findByLabelText(/Filter failing rule by git/));
+    fireEvent.click(
+      await screen.findByLabelText(/Filter failing rule by oncall/),
+    );
+    fireEvent.click(await screen.findByLabelText(/Filter exempt rule by k8s/));
+    expect(await screen.findByText(/Exempt Rule/)).toBeVisible();
```

Comment:
does checkForText not work here?

Reply:
One of the older comments in that test file said that `checkForText` didn't work on this page's tests so I didn't bother checking at first. But looks like it's now working.